src/main/java/edu/cornell/kfs/sys/service/impl/ISOFIPSConversionServiceImpl.java

```diff
@@ -0,0 +1,150 @@
+ *   No FIPS code exists                      AS = American Samoa
+ *   GB = Gabon                               GA = Gabon
+ *   UK = United Kingdom                      GB = United Kingdom of Great Britain and Northern Ireland (the)
+ *
+ *   BQ = NA VASSA ISLAND                     UM = United States Minor Outlying Islands (the)
+ *   DQ = JARVIS ISALND                       UM = United States Minor Outlying Islands (the)
+ *   FQ = BAKER ISLAND                        UM = United States Minor Outlying Islands (the)
+ *   HQ = HOWLAND ISLAND                      UM = United States Minor Outlying Islands (the)
+ *   JQ = JOHNSTON ATOLL                      UM = United States Minor Outlying Islands (the)
+ *   KQ = KINGMAN REEF                        UM = United States Minor Outlying Islands (the)
+ *   LQ = PALMYRA ATOLL                       UM = United States Minor Outlying Islands (the)
+ *   MQ = MIDWAY ISLANDS                      UM = United States Minor Outlying Islands (the)
+ *   WQ = WAKE ISLAND                         UM = United States Minor Outlying Islands (the)
+ *
+ *   AS = AUSTRALIA                           AU = Australia
+ *   AT = ASHMORE AND CARTIER ISLANDS         AU = Australia
+ *   CR = CORAL SEA ISLANDS TERRITORY         AU = Australia
+ *
+ *****************************************************************************/
+
+public class ISOFIPSConversionServiceImpl implements ISOFIPSConversionService {
+    
+    private static final Logger LOG = LogManager.getLogger(ISOFIPSConversionServiceImpl.class);
+    
+    private static final String NO_ISO_TO_FIPS_MAPPINGS_ERROR_MESSAGE = "No ISO-to-FIPS Country generic mapping found for ISO Country code : {0}";
+    private static final String MANY_ISO_TO_FIPS_MAPPINGS_ERROR_MESSAGE = "More than one ISO-to-FIPS Country generic mapping found for ISO Country code : {0}";
+    private static final String ONE_TO_ONE_ISO_TO_FIPS_MAPPING_MESSAGE = "One ISO-to-FIPS Country generic mapping found: ISO Country code : {0} mapped to FIPS Country code : {1}";
+    private static final String NO_FIPS_TO_ISO_MAPPINGS_ERROR_MESSAGE = "No FIPS-to-ISO Country generic mapping found for FIPS Country code : {0}";
+    private static final String MANY_FIPS_TO_ISO_MAPPINGS_ERROR_MESSAGE = "More than one FIPS-to-ISO Country generic mapping found for FIPS Country code : {0}";
+    private static final String ONE_TO_ONE_FIPS_TO_ISO_MAPPING_MESSAGE = "One FIPS-to-ISO Country generic mapping found FIPS Country code : {0} mapped to ISO Country code : {1}";
+
+    private ISOFIPSCountryMapDao isoFipsCountryMapDao;
+    
+    /**
+     * Return the single ACTIVE FIPS country code mapped to the ISO country code input parameter;
+     * Otherwise throw a NoISOtoFIPSMappingException or ManyISOtoFIPSMappingException.
+     *
+     * Active status of the MAPPING IS USED for this data retrieval.
+     */
+    public String convertISOCountryCodeToActiveFIPSCountryCode(String isoCountryCode) {
```

Comment:
You should put the "Override" annotation on the various service method implementations.

Reply:
adding annotation